parser/parser_test.go

```diff
@@ -546,37 +549,44 @@ else
 	testParseSuccess(t, "{%func (s *S) Foo(bar, baz string) %}{%endfunc%}")
 }
 
-func testParseFailure(t *testing.T, str string) {
+func testParseFailureWithDir(t *testing.T, str, dir string) {
 	r := bytes.NewBufferString(str)
 	w := &bytes.Buffer{}
-	if err := parse(w, r, "./foobar.tpl", "memory"); err == nil {
+	if err := Parse(w, r, filepath.Join(dir, "qtc-test.qtpl"), "memory"); err == nil {
 		t.Fatalf("expecting error when parsing %q", str)
 	}
 }
 
-func testParseSuccess(t *testing.T, str string) {
+func testParseSuccessWithDir(t *testing.T, str, dir string) {
 	r := bytes.NewBufferString(str)
 	w := &bytes.Buffer{}
-	if err := parse(w, r, "./foobar.tpl", "memory"); err != nil {
+	if err := Parse(w, r, filepath.Join(dir, "qtc-test.qtpl"), "memory"); err != nil {
 		t.Fatalf("unexpected error when parsing %q: %s", str, err)
 	}
 }
 
+func testParseFailure(t *testing.T, str string) {
+	testParseFailureWithDir(t, str, os.TempDir())
+}
+
+func testParseSuccess(t *testing.T, str string) {
+	testParseSuccessWithDir(t, str, os.TempDir())
+}
+
 func TestParseFile(t *testing.T) {
-	filename := "testdata/test.qtpl"
+	dir := filepath.Join(os.Getenv("GOPATH"), "src/github.com/valyala/quicktemplate/testdata/qtc")
```

Comment:
Use `build.Default.GOPATH` instead of `os.Getenv("GOPATH")`. See the explanation above